modular_skyrat/modules/vore/code/datums/components/vore.dm

```diff
@@ -0,0 +1,157 @@
+/datum/component/vore
+	var/list/obj/vbelly/bellies = list()
+	var/mob/living/owner
+	var/char_vars
+	var/vore_toggles = list()
+	var/selected_belly = 1
+	var/vore_enabled = TRUE
+	var/current_slot
+
+/datum/component/vore/Initialize()
+	if (!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	owner = parent
+	if (ishuman(owner))
+		var/mob/living/carbon/human/character = owner
+		current_slot = character.character_slot
+	RegisterSignal(owner, COMSIG_PARENT_PREQDELETED, .proc/handle_delete)
+	update_bellies(TRUE)
+
+/datum/component/vore/Destroy()
+	UnregisterSignal(owner, COMSIG_PARENT_PREQDELETED)
+	handle_delete()
+	. = ..()
+
+/datum/component/vore/proc/update_current_slot()
+	if (!isnull(current_slot))
+		return current_slot
+	if (ishuman(owner))
+		var/mob/living/carbon/human/character = owner
+		current_slot = character.character_slot
+		if (!isnull(current_slot))
+			return current_slot
+	current_slot = owner.client?.prefs.default_slot
+	return current_slot
+
+/datum/component/vore/proc/handle_delete()
+	SIGNAL_HANDLER
+
+	owner.release_belly_contents()
+
+/datum/component/vore/proc/get_belly_contents(bellynum, ref=FALSE, living=FALSE, as_string=FALSE, ignored=null, full=FALSE)
+	if (bellynum < 1 || bellynum > bellies.len)
+		return
+	var/list/belly_contents = bellies[bellynum].get_belly_contents(ref, living, as_string, ignored, full)
+	return belly_contents
+
+/datum/component/vore/proc/update_bellies(set_ref=FALSE) //update everything
+	if (!owner.client?.prefs?.vr_prefs)
+		return
+	var/datum/vore_prefs/vore = owner.client.prefs.vr_prefs
+	vore_toggles = vore.vore_toggles
+	char_vars = vore.char_vars
+	selected_belly = vore.selected_belly
+	vore_enabled = vore.vore_enabled
+	for (var/bellynum in 1 to vore.bellies.len)
+		var/belly_ref = (set_ref ? bellynum : null)
+		if (bellynum > bellies.len)
+			var/obj/vbelly/belly = new(null, owner, vore.bellies[bellynum], belly_ref)
+			bellies += belly
+		else
+			var/obj/vbelly/belly = bellies[bellynum]
+			belly.set_data(vore.bellies[bellynum], belly_ref)
+
+/datum/component/vore/proc/update_belly(bellynum, data)
+	if (!bellynum || bellynum > bellies.len)
+		return
+	bellies[bellynum].set_data(data)
+
+/datum/component/vore/proc/remove_belly(bellynum)
+	if (!bellynum || bellynum > bellies.len)
+		return
+	var/obj/vbelly/belly = bellies[bellynum]
+	belly.mass_release_from_contents()
+	bellies.Cut(bellynum, bellynum+1)
+	qdel(belly)
+
+//change this to mob/living once/if you make simplemob vore a thing
+/mob/living/carbon/human/Login()
+	. = ..()
+	if (!. || !client)
+		return
+	update_vore_verbs()
```

Comment:
ahaahhahaha nooooooooooooooooooooooooo absolutely not, no fucking with Login()

if your code needs to fuck with login you have failed on a design level, return to the drawing board and start from scratch